azure/services/virtualmachines/virtualmachines.go

```diff
@@ -111,6 +114,34 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 		s.Scope.SetAddresses(addresses)
 		s.Scope.SetVMState(infraVM.State)
+
+		spec, ok := vmSpec.(*VMSpec)
+		if !ok {
+			return errors.Errorf("%T is not a valid VM spec", vmSpec)
+		}
+		idsClient := identities.NewClient(s.Scope)
+
+		// Check if any userAssignedIdentities are missing from the VM and set a condition if so
```

Comment:
could we extract the logic checking for missing identities into a helper function and write a focused unit test for it?